src/microbe_stage/editor/tooltips/SelectionMenuToolTip.cs

```diff
@@ -213,12 +214,19 @@ public override void _ExitTree()
     /// <summary>
     ///   Instances the UI element for a modifier info
     /// </summary>
-    public void AddModifierInfo(string name, float value)
+    public void AddModifierInfo(string name, string value, float valueColor = 0, string? iconPath = null)
     {
+        // For stuff that loads too early
+        if (modifierInfoScene == null)
+            modifierInfoScene = GD.Load<PackedScene>("res://src/microbe_stage/editor/tooltips/ModifierInfoLabel.tscn");
```

Comment:
This shouldn't be like this. I'd say it'd be better to initialize this field in the constructor to ensure that this method works before adding nodes of this type to the scene tree.

Reply:
I encountered an error where modifierInfoScene was null for no clear reason to me. So I added this to ensure it's not

Reply:
Before you add a node to the scene tree, _Ready *will not be called* so if you call this method and *then* add the tooltip to the scene tree, this field will be uninitialized. But instead if the field is initialized in the constructor the field *cannot* be uninitialized when calling C# method irrespective whether the node is in the scene tree or not. I've written some node types that cannot be fully used before they are added to the scene tree. But I don't think such a requirement needs to be added here as a constructor early loading the required packed scene is probably sufficient.

Reply:
Oops appearently it was relevant to the unused Path variable

Reply:
The new direct node references are initialized earlier than when the old NodePath based lookup of child nodes happened. Could you maybe convert the entire selection menu class to use the direct references as you converted one of them and it wouldn't be that much more work to convert the rest (as there is an open issue about switching everything to use the new approach that is less error-prone).